docs/transpile/create-a-transpiler-plugin.ipynb

```diff
@@ -0,0 +1,433 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "58b1f0a7-d62e-4f71-ba01-b8f0beaeeb83",
+   "metadata": {},
+   "source": [
+    "# Create a transpiler plugin\n",
+    "\n",
+    "Creating a [transpiler plugin](transpiler-plugins) is a great way to share your transpilation code with the wider Qiskit community, allowing other users to benefit from the functionality you've developed. Thank you for your interest in contributing to the Qiskit community!\n",
+    "\n",
+    "Before you create a transpiler plugin, you need to decide what kind of plugin is appropriate for your situation. There are three kinds of transpiler plugins:\n",
+    "- [**Transpiler stage plugin**](/api/qiskit/transpiler_plugins). Choose this if you are defining a pass manager that can be substituted for one of the [6 stages](transpiler-stages) of a preset staged pass manager.\n",
+    "- [**Unitary synthesis plugin**](/api/qiskit/qiskit.transpiler.passes.synthesis.plugin.UnitarySynthesisPlugin). Choose this if your transpilation code takes as input a unitary matrix and outputs a description of a quantum circuit implementing that unitary.\n",
```

Comment:
To be super clear, maybe mention this unitary matrix is a `numpy` `array` as stated later on.

Reply:
Done.